src/main/java/com/eprosima/fastcdr/idl/templates/TypesSwigInterface.stg

```diff
@@ -62,6 +76,122 @@ $endif$
 %import(module="fastdds") "fastdds/dds/core/LoanableTypedCollection.hpp"
 %import(module="fastdds") "fastdds/dds/core/LoanableSequence.hpp"
 
+$if(ctx.thereIsInterface)$
+%include <exception.i>
+%include <std_shared_ptr.i>
+%include <std_pair.i>
+
+%import(module="fastdds") "fastdds/dds/rpc/exceptions/RpcException.hpp"
+%import(module="fastdds") "fastdds/dds/rpc/exceptions/RpcOperationError.hpp"
+
+%exception {
+    try
+    {
+        \$action
+    }
+    catch (const eprosima::fastdds::dds::rpc::RpcException& ex)
+    {
+        SWIG_exception(SWIG_RuntimeError, ex.what());
+    }
+    catch (const std::exception& ex)
+    {
+        SWIG_exception(SWIG_RuntimeError, ex.what());
+    }
+    catch (...)
+    {
+        SWIG_exception(SWIG_RuntimeError,"Unknown exception");
+    }
+}
+
+$if(ctx.thereIsOutputFeed)$
+%import(module="fastdds") "fastdds/dds/rpc/interfaces/RpcServerWriter.hpp"
+%ignore eprosima::fastdds::dds::rpc::RpcClientReader::read(T&);
+%ignore eprosima::fastdds::dds::rpc::RpcClientReader::read(T&,eprosima::fastdds::dds::Duration_t&);
+%import(module="fastdds") "fastdds/dds/rpc/interfaces/RpcClientReader.hpp"
+%extend eprosima::fastdds::dds::rpc::RpcClientReader {
+    std::pair<bool, T> read(
+        const eprosima::fastdds::dds::Duration_t& timeout = eprosima::fastdds::dds::c_TimeInfinite)
+    {
+        std::pair<bool, T> ret_val{};
+        if (eprosima::fastdds::dds::c_TimeInfinite == timeout)
+        {
+            ret_val.first = self->read(ret_val.second);
+        }
+        else
+        {
+            ret_val.first = self->read(ret_val.second, timeout);
+        }
+        return ret_val;
+    }
+}
+
+$ctx.outputFeedTypes : {feed_type | $output_feed(feed_type)$}; separator="\n\n"$
+$endif$
+
+$if(ctx.thereIsNonFeedOperation)$
+// Code for std::future taken from https://github.com/swig/swig/issues/1828#issuecomment-648449092
+namespace eprosima::fastdds::dds::rpc
+{
+template <class R>
+class RpcFuture {
```

Comment:
It is magic how this helper class for the swig wrapper (which is Fast is only an alias of std::future) links after with the implementation of `std::future` of C++ STL